kv/kvserver: TestLogGrowthWhenRefreshingPendingCommands failed

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/10a83506-7b7a-4ec9-91b3-46d58825d77e?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [bcc993d796d03664604bf695e38fd5644d0bc952](https://github.com/cockroachdb/cockroach/commits/bcc993d796d03664604bf695e38fd5644d0bc952):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1470519443
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2041: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1470519443
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (49.24s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (22.19s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2011: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"2","commit":105,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":105,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (6.03s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45044

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/031e68d1-20c1-46b8-b00f-93e9aa573a8e?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [bcc993d796d03664604bf695e38fd5644d0bc952](https://github.com/cockroachdb/cockroach/commits/bcc993d796d03664604bf695e38fd5644d0bc952):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands69067855
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2041: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands69067855
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (108.46s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (31.90s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2011: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"2","commit":58,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":58,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (9.42s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/5c9b0714-e971-432c-a162-bc020d586966?testReportRun=3&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [b1474fa887606008960634b571cf4501efb6281b](https://github.com/cockroachdb/cockroach/commits/b1474fa887606008960634b571cf4501efb6281b):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3167341054
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2041: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3167341054
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (38.02s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (12.81s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2011: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"3","commit":48,"lead":"3","leadEpoch":"2","raftState":"StateFollower","applied":48,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (4.32s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/b89582ed-f94a-4b03-804c-863887f2d977?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [b1474fa887606008960634b571cf4501efb6281b](https://github.com/cockroachdb/cockroach/commits/b1474fa887606008960634b571cf4501efb6281b):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3264654466
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2041: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3264654466
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (118.30s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (36.89s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2011: raft leader should be node 0, but got status {"id":"1","term":14,"vote":"2","commit":88,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":88,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (10.79s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/0b76d51f-77c2-440e-bdf5-dde4e96d25c4?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [749b563a7b5957b2db487c6e8e2fdeed8630103d](https://github.com/cockroachdb/cockroach/commits/749b563a7b5957b2db487c6e8e2fdeed8630103d):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands2021769156
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2041: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands2021769156
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (38.83s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (13.30s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2011: raft leader should be node 0, but got status {"id":"1","term":8,"vote":"2","commit":69,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":69,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (4.14s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/d295ec38-e4a6-4b7c-bdc1-a09ee1be59f9?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [749b563a7b5957b2db487c6e8e2fdeed8630103d](https://github.com/cockroachdb/cockroach/commits/749b563a7b5957b2db487c6e8e2fdeed8630103d):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1210514578
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2041: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1210514578
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (121.23s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (39.09s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2011: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"3","commit":84,"lead":"3","leadEpoch":"2","raftState":"StateFollower","applied":84,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (11.72s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/11802f68-0204-4eef-8f3d-9a7598f7085e?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1837229628
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2028: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1837229628
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (23.22s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (13.91s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:1998: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"3","commit":66,"lead":"3","leadEpoch":"2","raftState":"StateFollower","applied":66,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (4.97s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/292a8be9-004e-4f10-ae7d-1d648235cd0a?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3510702124
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2028: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3510702124
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (113.48s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:1998: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"2","commit":54,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":54,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (9.76s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (31.99s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/4f2e721e-604c-45c9-8aba-ee97798b6f04?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [44de2d379610067e14a7ebfbc92e64311f13a232](https://github.com/cockroachdb/cockroach/commits/44de2d379610067e14a7ebfbc92e64311f13a232):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1616585119
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2028: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1616585119
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (34.08s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (22.91s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:1998: raft leader should be node 0, but got status {"id":"1","term":11,"vote":"3","commit":87,"lead":"3","leadEpoch":"2","raftState":"StateFollower","applied":87,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (6.80s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/4fdd7866-a9cc-43a6-bc6e-4fdf98dffced?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [44de2d379610067e14a7ebfbc92e64311f13a232](https://github.com/cockroachdb/cockroach/commits/44de2d379610067e14a7ebfbc92e64311f13a232):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands4010935504
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2028: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands4010935504
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (109.39s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (32.17s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:1998: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"2","commit":69,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":69,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (9.28s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/5408450c-e644-4bdf-8ca9-9de3946835b0?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [844d7637f3b4dc1275e8aa05c1cf3bbb1f59f8eb](https://github.com/cockroachdb/cockroach/commits/844d7637f3b4dc1275e8aa05c1cf3bbb1f59f8eb):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands706599611
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands706599611
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (174.61s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (92.57s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"2","commit":71,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":71,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (11.57s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/e879b55b-74be-45c5-8361-9dc163ff2659?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [21d1123dce5401b1ec580d4156db5db075074506](https://github.com/cockroachdb/cockroach/commits/21d1123dce5401b1ec580d4156db5db075074506):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1635641813
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1635641813
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (22.37s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (13.28s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"4","commit":78,"lead":"4","leadEpoch":"2","raftState":"StateFollower","applied":78,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (4.26s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/73ee2d44-b3ce-4175-bff1-4c07407643b8?testReportRun=3&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [ccaed974f75032f1b908bca2d42145d21d0092ca](https://github.com/cockroachdb/cockroach/commits/ccaed974f75032f1b908bca2d42145d21d0092ca):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3137526485
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3137526485
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (21.04s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (11.96s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2025: raft log size grew to 43 KiB
    client_raft_test.go:2025: raft log size grew to 60 KiB
    client_raft_test.go:2025: raft log size grew to 60 KiB
    client_raft_test.go:2025: raft log size grew to 60 KiB
    client_raft_test.go:2025: raft log size grew to 60 KiB
    client_raft_test.go:2025: raft log size grew to 60 KiB
    client_raft_test.go:2025: raft log size grew to 60 KiB
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":6,"vote":"1","commit":62,"lead":"0","leadEpoch":"0","raftState":"StateFollower","applied":62,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (3.67s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/420ae188-840c-49d1-9a87-e0357b9e0aee?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [ccaed974f75032f1b908bca2d42145d21d0092ca](https://github.com/cockroachdb/cockroach/commits/ccaed974f75032f1b908bca2d42145d21d0092ca):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3245059797
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3245059797
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (105.77s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (31.43s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"3","commit":89,"lead":"3","leadEpoch":"2","raftState":"StateFollower","applied":89,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (9.30s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/40ee6bae-8fa2-49e5-8f89-576e9d363f7e?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [d7ea85402dc35e36c6cc35520fa91f25fd5c999d](https://github.com/cockroachdb/cockroach/commits/d7ea85402dc35e36c6cc35520fa91f25fd5c999d):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3893583432
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3893583432
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (99.58s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (27.57s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"2","commit":63,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":63,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (7.56s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/4c470b63-2c79-46b7-bf3e-230d0f04eaf9?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [b7331e4abaa1e9b10065c4e3b43b29c56dcb7f1d](https://github.com/cockroachdb/cockroach/commits/b7331e4abaa1e9b10065c4e3b43b29c56dcb7f1d):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1127192425
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands1127192425
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (166.59s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (92.33s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"2","commit":70,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":70,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (9.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/66171650-2667-4cee-b733-5bc20df13d04?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [5f83714a8111d0ea9ca8b34a72b2f5d97ea56b53](https://github.com/cockroachdb/cockroach/commits/5f83714a8111d0ea9ca8b34a72b2f5d97ea56b53):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands457876219
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands457876219
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (32.32s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (21.59s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":18,"vote":"2","commit":81,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":81,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (6.16s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/503a4c46-8661-4cf9-a66a-30f917be3110?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [5f83714a8111d0ea9ca8b34a72b2f5d97ea56b53](https://github.com/cockroachdb/cockroach/commits/5f83714a8111d0ea9ca8b34a72b2f5d97ea56b53):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands2685743987
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands2685743987
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (123.84s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (38.99s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":8,"vote":"2","commit":121,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":121,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (12.75s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/29be128b-bbb2-4b8c-89de-c0a0168fe550?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [d18eb683b2759fd8814dacf0baa913f596074a17](https://github.com/cockroachdb/cockroach/commits/d18eb683b2759fd8814dacf0baa913f596074a17):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3506725493
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3506725493
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (22.25s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (13.11s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"2","commit":60,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":60,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (4.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/928d754c-4f1f-4fd8-9fb8-50c3dec8a389?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [5b9cad925ece34f317e8785de719414bcf2aeb5a](https://github.com/cockroachdb/cockroach/commits/5b9cad925ece34f317e8785de719414bcf2aeb5a):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3379578077
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3379578077
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (174.87s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (92.37s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":10,"vote":"2","commit":112,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":112,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (9.62s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/a25ee585-e0ec-48d7-a8ad-515b1ecabec0?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [fada0424873ba279e5cfeb193d7d9d0b760693e3](https://github.com/cockroachdb/cockroach/commits/fada0424873ba279e5cfeb193d7d9d0b760693e3):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands475313134
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands475313134
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (102.63s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (30.00s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"3","commit":57,"lead":"3","leadEpoch":"2","raftState":"StateFollower","applied":57,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (9.75s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/dea60993-0098-4d0d-aed5-2915d429197e?testReportRun=3&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [fe048f6e72edca59b4367463db1d0d00783f08ab](https://github.com/cockroachdb/cockroach/commits/fe048f6e72edca59b4367463db1d0d00783f08ab):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands4181014688
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands4181014688
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (21.37s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2025: raft log size grew to 2.4 KiB
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":6,"vote":"1","commit":55,"lead":"0","leadEpoch":"0","raftState":"StateFollower","applied":55,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (3.59s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (12.35s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/2e7361fb-ab01-4290-8567-1fdc8b131565?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [a653b4e4e6483cec7d65808ba4d55d8c63747a6e](https://github.com/cockroachdb/cockroach/commits/a653b4e4e6483cec7d65808ba4d55d8c63747a6e):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3162548112
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands3162548112
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (23.08s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (13.91s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"4","commit":78,"lead":"4","leadEpoch":"2","raftState":"StateFollower","applied":78,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (5.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/3a4e1c20-183d-4309-84df-f24c95ab10c8?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [a653b4e4e6483cec7d65808ba4d55d8c63747a6e](https://github.com/cockroachdb/cockroach/commits/a653b4e4e6483cec7d65808ba4d55d8c63747a6e):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands214296146
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands214296146
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (244.61s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (149.71s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":16,"vote":"3","commit":184,"lead":"3","leadEpoch":"2","raftState":"StateFollower","applied":184,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (36.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/ecd927f7-fdb9-4756-a21c-37e0a0681ca6?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [d8de5b82dbbc107e15e59264cd8a48d4b98135cb](https://github.com/cockroachdb/cockroach/commits/d8de5b82dbbc107e15e59264cd8a48d4b98135cb):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands2842119509
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands2842119509
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (21.22s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (12.35s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":6,"vote":"1","commit":50,"lead":"0","leadEpoch":"0","raftState":"StateFollower","applied":50,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (3.82s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/e3181a20-fd67-4668-8c50-9a7380185c57?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [d8de5b82dbbc107e15e59264cd8a48d4b98135cb](https://github.com/cockroachdb/cockroach/commits/d8de5b82dbbc107e15e59264cd8a48d4b98135cb):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands4196671044
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands4196671044
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (107.65s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (31.42s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"2","commit":65,"lead":"2","leadEpoch":"1","raftState":"StateFollower","applied":65,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (9.14s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLogGrowthWhenRefreshingPendingCommands [failed](https://tanzanite.cluster.engflow.com/invocations/default/30a73318-514b-4256-892f-47a8d7a71f44?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [3b6f3c659f7c511761b44d65ba2cf62815a514c1](https://github.com/cockroachdb/cockroach/commits/3b6f3c659f7c511761b44d65ba2cf62815a514c1):


```
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands2538804827
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:2039: -- test log scope end --
test logs left over in: outputs.zip/logTestLogGrowthWhenRefreshingPendingCommands2538804827
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (129.01s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader
    --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader (39.49s)
=== RUN   TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true
    client_raft_test.go:2009: raft leader should be node 0, but got status {"id":"1","term":7,"vote":"3","commit":107,"lead":"3","leadEpoch":"2","raftState":"StateFollower","applied":107,"progress":{},"leadtransferee":"0"}
        --- FAIL: TestLogGrowthWhenRefreshingPendingCommands/lease-type=LeaseLeader/proposeOnFollower=true (12.52s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogGrowthWhenRefreshingPendingCommands.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/client_raft_test.go
+++ b/pkg/kv/kvserver/client_raft_test.go
@@ -1867,92 +1867,84 @@ func TestLogGrowthWhenRefreshingPendingCommands(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	testutils.RunValues(t, "lease-type", roachpb.TestingAllLeaseTypes(), func(t *testing.T, leaseType roachpb.LeaseType) {
-		ctx := context.Background()
-		settings := cluster.MakeTestingClusterSettings()
-		kvserver.OverrideDefaultLeaseType(ctx, &settings.SV, leaseType)
-
-		raftConfig := base.RaftConfig{
-			// Drop the raft tick interval so the Raft group is ticked more.
-			RaftTickInterval: 10 * time.Millisecond,
-			// Reduce the max uncommitted entry size.
-			RaftMaxUncommittedEntriesSize: 64 << 10, // 64 KB
-			// RaftProposalQuota cannot exceed RaftMaxUncommittedEntriesSize.
-			RaftProposalQuota: int64(64 << 10),
-			// RaftMaxInflightMsgs * RaftMaxSizePerMsg cannot exceed RaftProposalQuota.
-			RaftMaxInflightMsgs: 16,
-			RaftMaxSizePerMsg:   1 << 10, // 1 KB
-		}
-		// Suppress timeout-based elections to avoid leadership changes in ways this
-		// test doesn't expect. For leader leases, fortification itself provides us
-		// this guarantee.
-		if leaseType != roachpb.LeaseLeader {
-			raftConfig.RaftElectionTimeoutTicks = 1000000
-		}
-
-		const numServers int = 5
-		stickyServerArgs := make(map[int]base.TestServerArgs)
-		for i := 0; i < numServers; i++ {
-			stickyServerArgs[i] = base.TestServerArgs{
-				Settings: settings,
-				StoreSpecs: []base.StoreSpec{
-					{
-						InMemory:    true,
-						StickyVFSID: strconv.FormatInt(int64(i), 10),
-					},
-				},
-				RaftConfig: raftConfig,
-				Knobs: base.TestingKnobs{
-					Server: &server.TestingKnobs{
-						StickyVFSRegistry: fs.NewStickyRegistry(),
-					},
-					Store: &kvserver.StoreTestingKnobs{
-						// Disable leader transfers during leaseholder changes so that we
-						// can easily create leader-not-leaseholder scenarios.
-						DisableLeaderFollowsLeaseholder: true,
-						// Refresh pending commands on every Raft group tick instead of
-						// every RaftReproposalTimeoutTicks.
-						RefreshReasonTicksPeriod: 1,
-					},
-				},
-			}
-		}
-
-		tc := testcluster.StartTestCluster(t, numServers,
-			base.TestClusterArgs{
-				ReplicationMode:   base.ReplicationManual,
-				ServerArgsPerNode: stickyServerArgs,
-			})
-		defer tc.Stopper().Stop(ctx)
-		store := tc.GetFirstStoreFromServer(t, 0)
-
-		key := []byte("a")
-		tc.SplitRangeOrFatal(t, key)
-		tc.AddVotersOrFatal(t, key, tc.Targets(1, 2, 3, 4)...)
-
-		// Raft leadership is kept on node 0.
-		leaderRepl := store.LookupReplica(key)
-		require.NotNil(t, leaderRepl)
-
-		// Put some data in the range so we'll have something to test for.
-		incArgs := incrementArgs(key, 5)
-		if _, err := kv.SendWrapped(ctx, store.TestSender(), incArgs); err != nil {
-			t.Fatal(err)
-		}
-
-		// Wait for all nodes to catch up.
-		tc.WaitForValues(t, key, []int64{5, 5, 5, 5, 5})
-
 		// Test proposing on leader and proposing on follower. Neither should result
 		// in unbounded raft log growth.
 		testutils.RunTrueAndFalse(t, "proposeOnFollower", func(t *testing.T, proposeOnFollower bool) {
+			ctx := context.Background()
+			settings := cluster.MakeTestingClusterSettings()
+			kvserver.OverrideDefaultLeaseType(ctx, &settings.SV, leaseType)
 
-			// Restart any nodes that are down, we dont need to restart 2 since
-			// it started to finish the test case.
-			for _, s := range []int{3, 4} {
-				if tc.ServerStopped(s) {
-					require.NoError(t, tc.RestartServer(s))
+			raftConfig := base.RaftConfig{
+				// Drop the raft tick interval so the Raft group is ticked more.
+				RaftTickInterval: 10 * time.Millisecond,
+				// Reduce the max uncommitted entry size.
+				RaftMaxUncommittedEntriesSize: 64 << 10, // 64 KB
+				// RaftProposalQuota cannot exceed RaftMaxUncommittedEntriesSize.
+				RaftProposalQuota: int64(64 << 10),
+				// RaftMaxInflightMsgs * RaftMaxSizePerMsg cannot exceed RaftProposalQuota.
+				RaftMaxInflightMsgs: 16,
+				RaftMaxSizePerMsg:   1 << 10, // 1 KB
+			}
+			// Suppress timeout-based elections to avoid leadership changes in ways this
+			// test doesn't expect. For leader leases, fortification itself provides us
+			// this guarantee.
+			if leaseType != roachpb.LeaseLeader {
+				raftConfig.RaftElectionTimeoutTicks = 1000000
+			}
+
+			const numServers int = 5
+			stickyServerArgs := make(map[int]base.TestServerArgs)
+			for i := 0; i < numServers; i++ {
+				stickyServerArgs[i] = base.TestServerArgs{
+					Settings: settings,
+					StoreSpecs: []base.StoreSpec{
+						{
+							InMemory:    true,
+							StickyVFSID: strconv.FormatInt(int64(i), 10),
+						},
+					},
+					RaftConfig: raftConfig,
+					Knobs: base.TestingKnobs{
+						Server: &server.TestingKnobs{
+							StickyVFSRegistry: fs.NewStickyRegistry(),
+						},
+						Store: &kvserver.StoreTestingKnobs{
+							// Disable leader transfers during leaseholder changes so that we
+							// can easily create leader-not-leaseholder scenarios.
+							DisableLeaderFollowsLeaseholder: true,
+							// Refresh pending commands on every Raft group tick instead of
+							// every RaftReproposalTimeoutTicks.
+							RefreshReasonTicksPeriod: 1,
+						},
+					},
 				}
 			}
+
+			tc := testcluster.StartTestCluster(t, numServers,
+				base.TestClusterArgs{
+					ReplicationMode:   base.ReplicationManual,
+					ServerArgsPerNode: stickyServerArgs,
+				})
+			defer tc.Stopper().Stop(ctx)
+			store := tc.GetFirstStoreFromServer(t, 0)
+
+			key := []byte("a")
+			tc.SplitRangeOrFatal(t, key)
+			tc.AddVotersOrFatal(t, key, tc.Targets(1, 2, 3, 4)...)
+
+			// Raft leadership is kept on node 0.
+			leaderRepl := store.LookupReplica(key)
+			require.NotNil(t, leaderRepl)
+
+			// Put some data in the range so we'll have something to test for.
+			incArgs := incrementArgs(key, 5)
+			if _, err := kv.SendWrapped(ctx, store.TestSender(), incArgs); err != nil {
+				t.Fatal(err)
+			}
+
+			// Wait for all nodes to catch up.
+			tc.WaitForValues(t, key, []int64{5, 5, 5, 5, 5})
+
 			// Determine which node to propose on. Transfer lease to that node.
 			propIdx := 0
 			if proposeOnFollower {

